components/application-broker/internal/knative/subscription.go

```diff
@@ -0,0 +1,102 @@
+package knative
+
+import (
+	"log"
+	"regexp"
+
+	apicorev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/apis"
+	knapisv1alpha1 "knative.dev/pkg/apis/v1alpha1"
+
+	messagingv1alpha1 "knative.dev/eventing/pkg/apis/messaging/v1alpha1"
+)
+
+const (
+	// maxPrefixLength for limiting the max length of the name prefix
+	maxPrefixLength = 10
+
+	// generatedNameSeparator for adding a separator after the generated name prefix
+	generatedNameSeparator = "-"
+)
+
+type SubscriptionBuilder struct {
+	subscription *messagingv1alpha1.Subscription
+}
+
+func Subscription(prefix, namespace string) *SubscriptionBuilder {
```

Comment:
Please add a description for all public functions/methods

Reply:
Follow up PR: https://github.com/kyma-project/kyma/pull/6728/files